addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/ScanRulesUtils.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.commonlib;
+
+import java.util.regex.Pattern;
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.OutputDocument;
+import net.htmlparser.jericho.Source;
+import net.htmlparser.jericho.StartTag;
+import org.parosproxy.paros.network.HttpMessage;
+
+public final class ScanRulesUtils {
+    private ScanRulesUtils() {}
+
+    public static boolean isFont(HttpMessage msg) {
+        if (msg.getResponseHeader().hasContentType("font")) {
+            return true;
+        }
+        final Pattern PATTERN_FONT_EXTENSIONS =
+                Pattern.compile("(?:\\.ttf|\\.woff|\\.woff2|\\.otf)\\z", Pattern.CASE_INSENSITIVE);
+
+        String path = msg.getRequestHeader().getURI().getEscapedPath();
+        if (path != null) {
+            return PATTERN_FONT_EXTENSIONS.matcher(path).find();
+        }
+        return false;
+    }
+
+    public static boolean isJavaScript(HttpMessage msg) {
+        if (msg.getResponseHeader().isJavaScript()) {
+            return true;
+        }
+        final Pattern PATTERN_JS_EXTENSIONS =
+                Pattern.compile("(?:\\.js|\\.jsx)\\z", Pattern.CASE_INSENSITIVE);
+        String path = msg.getRequestHeader().getURI().getEscapedPath();
+        if (path != null) {
+            return PATTERN_JS_EXTENSIONS.matcher(path).find();
+        }
+        return false;
+    }
+
+    public static boolean isCss(HttpMessage msg) {
+        return msg.getResponseHeader().isCss() || msg.getRequestHeader().isCss();
+    }
+
+    public static boolean isControlCharacter(HttpMessage msg) {
```

Comment:
Specifically `reaponseContainsControlChars` and you only need to check the response